app/helpers/calendar_helper.rb

```diff
@@ -0,0 +1,24 @@
+module CalendarHelper
+
+  def get_event(day, time)
+    datetime = add_date_and_time(day, time)
+    @events[day.wday].find do |event|
+      event.start_time == datetime
+    end if @events[day.wday]
+  end
+
+  def event_ongoing?(day, time)
+    datetime = add_date_and_time(day, time)
+    @events[day.wday].any? do |event|
+      event.start_time < datetime && event.end_time > datetime
+    end if @events[day.wday]
+  end
+
+  def event_span(event)
+    event ? (event.end_time - event.start_time) / 30.minutes : 1
```

Comment:
Is there right now a 'silent' assumption that every event's duration is a multiple of 30 min?